_posts/2019-06-21-install-esxi-using-foreman-using-ipxe-bootstrapping.md

```diff
@@ -0,0 +1,209 @@
+---
+layout: post
+title: 'Install ESXi using Foreman using iPXE bootstrapping'
+date: 2019-06-21 14:25:45
+author:
+  name: Amardeep Kahali
+tags:
+- foreman
+- esxi
+- vmware
+- ipxe
+- bootstrapping
+- uefi
+guest_post: true
+---
+
+This post is intended to simplify the ESXi installation process [described in the previous post](https://theforeman.org/2018/08/deploying-esxi-through-foreman.html) using the new iPXE bootstrapping feature of Foreman introduced in v1.20.
+
+### Requirements
+
+* A fully working Foreman instance. Follow the [manual](https://theforeman.org/manuals/latest/index.html#3.InstallingForeman) to install Foreman. After installation, please configure Foreman so that a Linux OS (for e.g. Centos) can be deployed successfully through Foreman.
+
+* An [ESXi ISO](https://my.vmware.com/en/group/vmware/evalcenter). For this demo we will use `VMware-VMvisor-Installer-6.7.0-8169922.x86_64.iso` but the same process will work on any version of ESXi.  
+
+Note: This demo is run on a Ubuntu system. 
+
+### 1: Enable iPXE in Foreman
+
+---
+
+Since `foreman-installer` doesn't support creating an iPXE based Foreman instance yet, we need to make some changes to the DHCP configuration manually to enable iPXE. The `dhcpd.conf` should look like this:
+
+    ```
+    # Bootfile Handoff
+    next-server ${foreman_server_fqdn};
```

Comment:
This does not appear to be valid config syntax, does it? Is this a pseudo-variable? Then we should tell.

Also `next-server` is preferred to be an IP address than hostname.

Reply:
Alright! Will make next-server to be an IP address variable